lib/config.js

```diff
@@ -66,13 +66,14 @@ class Config {
         this.parser = options.parser;
         this.parserOptions = options.parserOptions || {};
 
+        this.configCache = new ConfigCache();
```

Comment:
I don't understand the issue -- the test asserts that a specific value `rules` and `env` is present, which is the expected value for extending the given configs.

The `extends` are still present on the `baseConfig` due to https://github.com/eslint/eslint/pull/8636 (we don't remove `extends` from a merged config even after applying the extensions).